doc/interoperability/facade-types.md

```diff
@@ -223,19 +223,80 @@ Inner classes and objects will be looked up as fields of the enclosing JS object
-considered to represent the global scope.
+functions. Instead, in Scala.js, top-level objects annotated with
+`@JSGlobalScope` are considered to represent the global scope.
 
 {% highlight scala %}
+import js.annotation._
+
 @js.native
-object DOMGlobalScope extends js.GlobalScope {
+@JSGlobalScope
+object DOMGlobalScope extends js.Object {
   val document: HTMLDocument = js.native
 
   def alert(message: String): Unit = js.native
 }
 {% endhighlight %}
 
+Prior to 0.6.13, `extends js.GlobalScope` was used instead of `@JSGlobalScope`.
+`js.GlobalScope` is now deprecated.
+
+## <a name="import"></a> Imports from other JavaScript modules
+
+**Important:** Importing from JavaScript modules requires that you [emit a module for the Scala.js code](../project/module.html).
+
+The previous sections on native classes and objects all load things from the JavaScript global scope (through zero or more property accesses from there).
+In modern JavaScript ecosystems, we often want to load things from other *modules*.
+This is what `@JSImport` is designed for.
+You can annotate an `@js.native` class or object with `@JSImport` to signify that it is defined in a module.
+For example, in the following snippet:
+
+{% highlight scala %}
+@js.native
+@JSImport("bar.js", "Foo")
+class Foobaz(val x: Int) extends js.Object
+
+val f = new Foobaz(5)
+{% endhighlight %}
+
+the annotation specifies that `Foobaz` is a native JS class defined in the module `"bar.js"`, and exported under the name `"Foo"`.
+Semantically, `@JSImport` corresponds to an [ECMAScript 2015 import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), and the above code is therefore equivalent to this JavaScript code:
+
+{% highlight javascript %}
+import { Foo as Foobaz } from "bar.js";
+var f = new Foobaz(5);
+{% endhighlight %}
+
+In CommonJS terms, this would be:
+
+{% highlight javascript %}
+var bar = require("bar.js");
+var f = new bar.Foo(5);
+{% endhighlight %}
+
+The first argument to `@JSImport` is the name of the JavaScript module you wish to import.
+The second argument denotes what member of the module you are importing.
+It can be one of the following:
+
+* A string indicating the name of member.
+  The string can be a `.`-separated chain of selections (e.g., `"Foo.Babar"`).
+* The constant `JSImport.Default`, to select the *default* export of the JavaScript module.
+  This corresponds to `import Foobaz from "bar.js"`.
+* The constant `JSImport.Namespace`, to select the module itself (with its exports as fields).
+  This corresponds to `import * as Foobaz from "bar.js"`.
+
+The latter is particularly useful if you want to import members of the modules that are neither classes nor objects (for example, functions):
+
+{% highlight scala %}
+@js.native
+@JSImport("bar.js", JSImport.Namespace)
+object Bar {
+  def exportedFunction(x: Int): Int = js.native
+}
+
+val y = Bar.exportedFunction(5)
+{% endhighlight %}
+
```

Comment:
You should document how these map to CommonJS constructs. The only place this is currently documented is inside the compiler.
The choice between default and namespace is particularly involved. It essentially requires the developer to guess, how the module is going to export its members in an ES2015 module:

- If the whole module is moving to the default export, use `JSImport.Default`. This makes sense for modules exporting a single class. For example:

    ```javascript
    var JSZip = require("jszip.js");
    var zip = new JSZip();
    ```
- If the individual members will be exported, use `JSImport.Namespace`. For example:

    ```javascript
    var _ = require("underscore.js");
    _.each(...);
    ```

It is important to understand that this is only a guess (if there is no ES2015 version of the module yet) or must be based on how this particular module exports itself to CommonJS vs ES2015.

This becomes of course only relevant, once we actually support emitting ES2015 modules. However, it is important in the documentation now, to understand the difference between `JSImport.Default` and `JSImport.Namespace`, because there is no difference between them for most CommonJS modules.